src/twisted/trial/test/test_deferred.py

```diff
@@ -133,6 +133,15 @@ def test_passInlineCallbacks(self):
         self.assertEqual(result.testsRun, 1)
         self.assertTrue(detests.DeferredTests.touched)
 
+    def test_passCoroutineFunction(self):
+        """
+        The body of an async def test gets run.
```

Comment:
something about the reactor here, maybe? completes when the coroutine runs to completion, runs with `Deferred.fromCoroutine` or something, i.e. it's not expecting a `Future`

Reply:
The test_passCoroutineFunction traps with a completed Deferred and so never interacts with the reactor